nodejs/awsx/apigateway/authorizer.ts

```diff
@@ -0,0 +1,114 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// These APIs are currently experimental and may change.
+
+import * as aws from "@pulumi/aws";
+import * as pulumi from "@pulumi/pulumi";
+
+import * as awslambda from "aws-lambda";
+
+export type AuthorizerEvent = awslambda.CustomAuthorizerEvent;
+export type AuthorizerResponse = awslambda.CustomAuthorizerResult;
+
+/**
+ * SecurityDefinition provides the definition for a custom Authorizer for API Gateway.
+ */
+export interface SecurityDefinition {
+    /**
+     * Pretty name for the security definition to be referenced as. This is only used within the
+     * swagger and must be unique for each unique authorizer within the API. If you want
+     * to share an authorizer across routes, provide the same name for them. If not defined, a unique
+     * name for you.
+     */
+    authorizerName?: string;
+
+    /**
+     * parameterName is the name of the header or query parameter containing the authorization token. Must be
+     * "Unused" for multiple identity sources or non header or query type of request parameters.
+     * */
+    parameterName: string;
+
+    /**
+     * Required and the value must be "header" or "query" for an API Gateway API. Must be "header" for multiple
+     * identity sources or non header or query type of request parameters.
```

Comment:
i'm finding the sentences here a bit hard to parse.  note: stating that it's required isn't really necessary given that hte property isn't optional.  also, stating it has to eb 'header' or 'query' is shown in the type signature.  so it might be better to explain what those two values mean instead.

Reply:
I will re-read / re-write these